examples/load_from_cloud.html

```diff
@@ -0,0 +1,474 @@
+    const expiration = 12 * 60 * 60; // accessible for 12 hours
+
+    const canonicalQueryString =
+    'X-Goog-Algorithm=' + algorithm +
+    '&X-Goog-Credential=' + encodeURIComponent(clientEmail) +
+    '%2F' + dateDay + '%2Fauto%2Fstorage%2Fgoog4_request' +
+    '&X-Goog-Date=' + dateWithSecondsFormatted +
+    '&X-Goog-Expires=' + expiration.toString() +
+    '&X-Goog-SignedHeaders=host' +
+    (!headers? "" : '%3Brange');
+
+    const canonicalRequest =
+    'GET\n' +
+    pathAfterBucket + '\n' +
+    canonicalQueryString + '\n' +
+    'host:potree-files.storage.googleapis.com\n' +
+    (!headers? '\n' : 'range:' + headers.Range + '\n\n') +
+    (!headers? 'host' : 'host;range') + '\n' +
+    'UNSIGNED-PAYLOAD';
+
+    const encodedCanonical = await encodeCanonical(canonicalRequest);
+    const stringToSign = `${algorithm}\n${dateWithSecondsFormatted}\n${dateDay}/auto/storage/goog4_request\n${encodedCanonical}`;
+
+    // RSA signature generation
+    const signatureBuffer = await rsaSha256(privateKey, stringToSign);
+    const signature = buf2hex(signatureBuffer);
+
+    const signedUrl = `https://${bucket}.storage.googleapis.com${pathAfterBucket}?${canonicalQueryString}&x-goog-signature=${signature}`;
+    return signedUrl;
+  }
+
+  async function runGCPExample() {
+    const BUCKET = document.getElementById("gcp_bucket").value;
+    const NAME = document.getElementById("name").value;
+    const PATH = document.getElementById("path").value;
+    const KEY_FILE = document.getElementById("gcp_key_file").files[0];
+    document.getElementById("login-form").style.display = "none";
+    document.getElementById("potree-container").style.display = "initial";
+  	window.viewer = new Potree.Viewer(document.getElementById("potree_render_area"));
+
+  	viewer.setEDLEnabled(true);
+  	viewer.setFOV(60);
+  	viewer.setPointBudget(1_000_000);
+  	viewer.loadSettingsFromURL();
+
+  	viewer.setDescription("");
+
+  	viewer.loadGUI(() => {
+  		viewer.setLanguage('en');
+  		$("#menu_appearance").next().show();
+  		//viewer.toggleSidebar();
+  	});
+
+    // read in the key file given by user
+    const fr = new FileReader();
+    fr.onload = async function() {
+      const result = fr.result;
+      const signUrlGCP =
+      async (url, headers) => getGCPSignedUrl(url, headers, result, BUCKET);
+
+      const e = await Potree.loadPointCloud(BUCKET + "/" + PATH, NAME, null, signUrlGCP);
+      viewer.scene.addPointCloud(e.pointcloud);
+      const material = e.pointcloud.material;
+      material.size = 1;
+      material.pointSizeType = Potree.PointSizeType.ADAPTIVE;
+
+      viewer.fitToScreen();
+    }
+
+    try {
+      fr.readAsText(KEY_FILE);
+    } catch(e) {
+      throw("Given file is incompatible.");
+    }
+  }
+
+  const gcpButton = document.getElementById("gcp-start-button");
+  gcpButton.addEventListener('click', runGCPExample);
+
+
+</script>
+  </body>
+</html>
```

Comment:
Needs newline at the end of the line. (There is probably a setting in your editor that will enforce this.)